.github/workflows/install_prerequisites.cmake

```diff
@@ -0,0 +1,47 @@
+message(STATUS "Using host CMake version: ${CMAKE_VERSION}")
+
+if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows")
```

Comment:
This stuff would probably be easier to do in the GHA workflow file

Reply:
You can only run scripts via the YAML file AFAIK, but not really do different things in the run command if using a matrix configuration. With scripts, you'd need to use bash on Linux/Mac and PowerShell for Windows. CMake is platform-agnostic and has a shell/script mode I'm using here. It's different from the standard configure/build usage as it only runs a CMake script without configuring any build system etc. [See the CMake docs](https://cmake.org/cmake/help/latest/manual/cmake.1.html#run-a-script) for more on that topic.

Another solution would be having one workflow per platform. In this case might not be so different as there's not much commonality in the configuration/install calls due to platform specifics.